base_tier_validation_server_action/models/tier_review.py

```diff
@@ -1,20 +1,24 @@
 # Copyright 2020 Ecosoft (http://ecosoft.co.th)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import api, models
+from odoo import models
 
 
 class TierReview(models.Model):
     _inherit = "tier.review"
 
-    @api.constrains("status")
-    def _trigger_server_action(self):
-        for rec in self.filtered(lambda l: l.status in ["approved", "rejected"]):
-            server_action = False
-            if rec.status == "approved":
-                server_action = rec.definition_id.server_action_id
-            if rec.status == "rejected":
-                server_action = rec.definition_id.rejected_server_action_id
-            if server_action:
-                ctx = {"active_model": rec.model, "active_id": rec.res_id}
-                server_action.with_context(ctx).run()
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get("status") in ["approved", "rejected"]:
+            for rec in self:
+                server_action = False
+                if rec.status == "approved":
+                    server_action = rec.definition_id.server_action_id
+                if rec.status == "rejected":
```

Comment:
```suggestion
                elif rec.status == "rejected":
```
if status is approved, it can't be rejected.

Reply:
I think the result is the same, status can never be both either.
Note: I am trying to not use `else` and `elif` also in the code, the code will be cleaner.